app/helpers/icons.py

```diff
@@ -14,3 +17,16 @@ def get_icon_template_url(base_url='', with_color=True):
         color_part = "-{r},{g},{b}"
     return f"{get_icon_set_template_url(base_url)}/icons/{{icon_name}}" \
            f"@{{icon_scale}}{color_part}.png"
+
+
+def calculate_icon_size(size, icon_set, scale=1):
+    """
+    Calculate icon size so that the size of the smaller dimension is equal to the standard icon size
+    multiplied by the scaling
+    """
+    width, height = size
+    min_size = min(width, height)
+    return (
+        int(scale * DEFAULT_ICON_SIZE * width / min_size),
+        int(scale * DEFAULT_ICON_SIZE * height / min_size)
+    )
```

Comment:
This will set the size of the smaller dimension of the icon to the standard size (48 px) scaled by the scale factor so that we have a unified size for the icons